File: Python_BOJ/2312.py
import sys
n = int(sys.stdin.readline().rstrip())

def show(num):
    count = [0 for _ in range(num + 1)]
    temp = num
    for i in range(2, num + 1):
        while temp % i == 0:
            count[i] += 1
            temp /= i
    for i in range(2, num + 1):
        if count[i] > 0:
            print(i, count[i])

for _ in range(n):
    num = int(sys.stdin.readline().rstrip())
    show(num)

File: Python_BOJ/2312.c
#include <stdio.h>
#include <stdlib.h>

void show(int num) {
	int* count = (int*)calloc((num + 1), sizeof(int));
	int temp = num;
	for (int i = 2; i <= num; i++) {
		while (temp % i == 0) {
			count[i]++;
			temp /= i;
		}
	}

	for (int i = 2; i <= num; i++) {
		if (count[i] > 0) {
			printf("%d %d\n", i, count[i]);
		}
	}
	free(count);
}

int main(void) {

	int n = 0;
	scanf("%d", &n);
	while (n--) {
		int num = 0;
		scanf("%d", &num);
		show(num);
	}

	return 0;
}